modules/cats/src/main/scala/pureconfig/module/cats/package.scala

```diff
@@ -20,6 +21,13 @@ package object cats {
         case Some(nonEmpty) => Right(nonEmpty)
       }
     }
+  private[pureconfig] def fromNonEmpty2[F[_, _], G[_, _], A, B](fromFT: F[A, B] => Option[G[A, B]])(cur: ConfigCursor)(implicit ct: ClassTag[F[A, B]], fReader: ConfigReader[F[A, B]]): Either[ConfigReaderFailures, G[A, B]] =
+    fReader.from(cur).right.flatMap { ft =>
+      fromFT(ft) match {
+        case None ⇒ cur.failed(EmptyTraversableFound(ct.toString))
+        case Some(nonEmpty) ⇒ Right(nonEmpty)
+      }
```

Comment:
Way too complicated, IMO. Something like this should work for all `NonEmpty*` types:
```scala
private[pureconfig] def fromNonEmpty[X, Y](reader: ConfigReader[X])(fromX: X => Option[Y])(
  implicit ct: ClassTag[X]
): ConfigReader[Y] =
  reader.emap(x => fromX(x).toRight(EmptyTraversableFound(ct.toString)))

implicit def nonEmptyListReader[T](implicit reader: ConfigReader[List[T]]): ConfigReader[NonEmptyList[T]] =
  fromNonEmpty(reader)(NonEmptyList.fromList)

implicit def nonEmptyMapReader[A, B](implicit reader: ConfigReader[Map[A, B]], ord: Order[A]): ConfigReader[NonEmptyMap[A, B]] =
  fromNonEmpty(reader)(x => NonEmptyMap.fromMap(SortedMap(x.toSeq: _*))
```
(importing `cats.syntax.option._`)

Reply:
This indeed looks better.
But I've tried to compile the code you've provided but I'm getting errors.
>[error] ClassTag available for cats.data.NonEmptyMap[A,B]
[error]     fromNonEmpty(reader)(x => NonEmptyMap.fromMap(SortedMap(x.toSeq: _*)))

Any idea how to fix this ?

Reply:
I see the problem now. Basically, `NonEmptyMap` is not a class/interface, but a type with some constraints, meaning that summoning a `ClassTag` for it is not possible. Reverting to `ct: ClassTag[X]` in `fromNonEmpty` should do the trick. This would be less precise, but it was good enough for the existing implementation...

Otherwise we can switch to `scala.reflect.runtime.universe.WeakTypeTag`. However, I don't remember the specifics of the `ClassTag`/`TypeTag`/`WeakTypeTag` distinction well enough to make any predictions about what this will an will not break.

Reply:
Ok, this works thanks.

I've pushed the amendments as per PR comments.